### PR TITLE
Android x

### PR DESCRIPTION
**Pull request by jordan1997**
Add support for android x
Comment for authors:
I added android-x artifact into the build.gradle for the module, but you need to check this.

P.S. all suggestion will check and resolve :)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,16 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
+    ext.kotlin_version = '1.2.51'
```
Comment (senneco) [inline build.gradle]: I think it's not relate to current issue

File: build.gradle [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+
     repositories {
         jcenter()
         mavenCentral()
         google()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.2'
-    }
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version" }
```
Comment (senneco) [inline build.gradle]: I think it's not relate to current issue

File: gradle.properties [hunk 3]
```diff
@@ -17,4 +17,6 @@
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 
-#org.gradle.java.home=<path to jdk1.8 for tests>
\ No newline at end of file
+#org.gradle.java.home=<path to jdk1.8 for tests>
+android.useAndroidX=true
```
Comment (senneco) [inline gradle.properties]: I think it's not relate to current issue

Comment (jordan1997) [reply gradle.properties]: Hi, @senneco

File: build.gradle (hunk 1 above)
Comment (jordan1997) [reply build.gradle]: I check and update PR in several hours.

File: build.gradle (hunk 2 above)
Comment (jordan1997) [reply build.gradle]: I check and update PR in several hours.

File: gradle.properties (hunk 3 above)
Comment (senneco) [reply gradle.properties]: But this not related to moxy, right?) Could it be placed to target moxy-andoridx-sample module?

Comment (jordan1997) [reply gradle.properties]: Sorry, my mistake
I didn't think about another gradle.properties file in for the one module.

**Comment by jordan1997** [comment]
@senneco  I check your comments and solve them. And add the small fix to the sample custom strategy

File: sample-custom-strategy/src/main/kotlin/com/redmadrobot/app/presentation/launch/LaunchFragment.kt [hunk 4]
```diff
@@ -49,11 +49,11 @@ class LaunchFragment : MvpAppCompatFragment(), LaunchView {
         return LaunchPresenter()
     }
 
-    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        return inflater?.inflate(R.layout.fragment_launch, container, false)
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return inflater.inflate(R.layout.fragment_launch, container, false)!!
```
Comment (AlexeyKorshun) [inline sample-custom-strategy/src/main/kotlin/com/redmadrobot/app/presentation/launch/LaunchFragment.kt]: We can return null for non graphical fragment.
From documentation:
> Called to have the fragment instantiate its user interface view. This is optional, and non-graphical fragments can return null (which is the default implementation)

File: moxy-androidx/build.gradle [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+def getBasePom(String libName) {
+
+    def basePom = pom {
+        //noinspection GroovyAssignabilityCheck
+        project {
+            groupId targetGroupId
+            artifactId targetArtefactId
+            version targetVersionName
+            name 'Moxy-app-compat'
```
Comment (senneco) [inline moxy-androidx/build.gradle]: There should be 'Moxy-androidx'

Comment (jordan1997) [reply moxy-androidx/build.gradle]: Fixed

**Comment by ahulyk** [comment]
@senneco when that can be merged?

**Comment by ahulyk** [comment]
i did own version ;)

**Pull request merged**